src/redux/wallets/actions/wallet-actions.ts

```diff
@@ -778,3 +787,45 @@ export const addPublishedTxToAccount = (
         }
     });
 };
+
+export const setWalletPublicKey = (walletId: string, walletPublicKey: string) => {
+    return {
+        type: SET_WALLET_PUBLIC_KEY,
+        data: { walletId, walletPublicKey }
+    };
+};
+
+export const setWalletsCredentials = (password: string) => async (
+    dispatch: Dispatch<any>,
+    getState: () => IReduxState
+) => {
+    const state = getState();
+
+    for (const wallet of Object.values(state.wallets)) {
+        try {
+            const walletId = (wallet as IWalletState).id;
+
+            const storageWallet = await HDWallet.loadFromStorage(walletId, password);
+            const walletCredentials = storageWallet.getWalletCredentials();
+
+            if (walletCredentials) {
+                store.dispatch(setWalletPublicKey(walletId, walletCredentials.publicKey));
```

Comment:
you can use 

` setWalletPublicKey(walletId, walletCredentials.publicKey)(dispatch, getState);`